src/main/kotlin/Read.kt

```diff
@@ -0,0 +1,96 @@
+import com.mongodb.client.model.Filters
+import com.mongodb.client.model.Filters.eq
+import com.mongodb.client.model.IndexOptions
+import com.mongodb.client.model.Indexes
+import com.mongodb.client.model.Sorts
+import com.mongodb.kotlin.client.coroutine.MongoDatabase
+import kotlinx.coroutines.runBlocking
+
+fun main() {
+    val db = setupConnection()
+
+    runBlocking {
+        readAnyDocument(database = db)
+        readSpecificDocument(database = db)
+        readWithPaging(database = db, offset = 2, pageSize = 2)
+        readWithIndex(database = db)
+    }
+}
+
+suspend fun readAnyDocument(database: MongoDatabase) {
+    val collection = database.getCollection<Restaurant>(collectionName = "restaurants")
+    collection.find<Restaurant>().limit(1).collect {
+        println(it)
+    }
+}
+
+suspend fun readSpecificDocument(database: MongoDatabase) {
+    val collection = database.getCollection<Restaurant>(collectionName = "restaurants")
+    val queryParams = Filters
+        .and(
+            listOf(
+                eq("cuisine", "American"),
+                eq("borough", "Queens")
+            )
+        )
+
+
+    collection
+        .find<Restaurant>(queryParams)
+        .limit(2)
+        .collect {
+            println(it)
+        }
+
+}
+
+suspend fun readWithPaging(database: MongoDatabase, offset: Int, pageSize: Int) {
+    val collection = database.getCollection<Restaurant>(collectionName = "restaurants")
+    val queryParams = Filters
+        .and(
+            listOf(
+                eq(Restaurant::cuisine.name, "American"),
+                eq(Restaurant::borough.name, "Queens")
+            )
+        )
+
+    collection
```

Comment:
From @johnlpage : Paging with Skip and Limit is pretty controversial too although it's almost OK here.

Reply:
Noted, added a note for the same in the blog.